nifi-ignite-extensions/nifi-ignite-services/src/main/java/pl/touk/nifi/services/IgniteLookupService.java

```diff
@@ -0,0 +1,144 @@
+package pl.touk.nifi.services;
+
+import org.apache.ignite.binary.BinaryObject;
+import org.apache.ignite.binary.BinaryType;
+import org.apache.ignite.internal.binary.BinaryUtils;
+import org.apache.ignite.internal.binary.GridBinaryMarshaller;
+import org.apache.nifi.annotation.lifecycle.OnEnabled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.flowfile.attributes.CoreAttributes;
+import org.apache.nifi.lookup.LookupFailureException;
+import org.apache.nifi.lookup.RecordLookupService;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.serialization.SimpleRecordSchema;
+import org.apache.nifi.serialization.record.*;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/*
+    TODO:
+    1. handle all ignite types
+    2. more test cases
+*/
+public class IgniteLookupService extends AbstractIgniteCache<String, BinaryObject> implements RecordLookupService {
```

Comment:
I think we need to mention somewhere (docs?) assumptions about cache e.g. that it is handled as BinaryObject, so we assume it's more or less SQL-based